packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -159,13 +166,20 @@ export class GraphQLRunner {
     }
 
     const document = this.parse(query)
-    const errors = this.validate(schema, document)
+    const { errors, warnings } = this.validate(schema, document)
 
     // Queries are usually executed in batch. But after the batch is finished
     // cache just wastes memory without much benefits.
     // TODO: consider a better strategy for cache purging/invalidation
     this.scheduleClearCache()
 
+    if (warnings.length > 0) {
+      warnings.forEach(err => {
+        const message = context.path ? `\nQueried in ${context.path}` : ``
```

Comment:
Few notes on this message:

1. For page/static queries at least I think we treat validation we return few lines below in somewhat richer way - https://github.com/gatsbyjs/gatsby/blob/4638cd678b28f7a515df465be21d9d8bcdd71d9a/packages/gatsby/src/query/query-runner.ts#L48-L86

While we can't really use "structured" warning (because only errors are structured really), maybe we could use what we already have there to have richer message - particularly `codeFrame` part that would help users to find those deprecations easier. As we use https://babeljs.io/docs/en/babel-code-frame we could even just pass ~error~ warning message as message to `codeFrameColumns` function.

In this function we don't really have as rich context as in `query-runner`, so maybe we should pass warnings back and let `graphql-runner` (for page/static queries) and https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/create-graphql-runner.ts for gatsby-node queries to show richer warnings?

Not something I want to push, just want to put idea out there

2. `context.path` here kind of works for page queries, but static queries will have either hash or `sq--<something>` (don't remember now which one is it :P) so that information might bring very little value to users. `gatsby-node` ones will have no path so that's even less info for users to figure out where something is happening

Reply:
Oh, and to clarify my comment - I think we can merge as-is without doing rich stuff just yet (potential future improvement), but we should at least check interaction with static queries to see how message looks like with current code

Reply:
100% agree that we need a follow-up and move those deprecations to the caller side of this graphql runner.

As for static queries, they were displayed like this:

```
warn The field Author.summary is deprecated. Use name instead.
Queried in sq--src-components-bio-js
```

I can add a special case and format this path to something nicer, e.g.:
```
warn The field Author.summary is deprecated. Use name instead.
Queried in src/components/bio.js
```

I actually thought about this originally just didn't want to mess with this value as it is supposed to be opaque.

Reply:
After some thinking, I've added this formatting for static query ids. At least will be less confusing for users until we actually move it to `query-runner.ts`.

Reply:
Should we just pass extra metadata from `query-runner` that point to actual file (`queryJob.componentPath`) and just not even try relying on `context.path`? Given that `context.path` is available only in page/static queries anyway, that's only code path that would result in `Queried in X` adnotation

Reply:
Great suggestion. Thanks! Updated in https://github.com/gatsbyjs/gatsby/pull/29450/commits/081bf8f0accd888ee166829928ebc92cfa0c0bd2

![image](https://user-images.githubusercontent.com/115628/107809125-33209b00-6d9d-11eb-8f65-5819f439ad9d.png)